### ISSUE TITLE
class load error when updating the plugin in a local install

### ISSUE THREAD

**Issue by docteurklein** [hirak/prestissimo#46]
If I re-run the `composer update` a secnd time, the problem disappears.
It seems logical since the same process is used to load the class that will be used later.

I think this is only due to the fact I installed it locally.

``` 
 % composer update
Loading composer repositories with package information
Updating dependencies (including require-dev)
  - Removing hirak/prestissimo (0.1.8)
  - Installing hirak/prestissimo (0.1.9)
    Downloading: Connecting...PHP Fatal error:  Uncaught Error: Class 'Hirak\Prestissimo\OutputFile' not found in /home/florian/workspace/qspot/vendor/hirak/prestissimo/src/CurlRemoteFilesystem.php:73
Stack trace:
#0 /home/florian/workspace/qspot/vendor/hirak/prestissimo/src/CurlRemoteFilesystem.php(173): Hirak\Prestissimo\CurlRemoteFilesystem->Hirak\Prestissimo\{closure}(Resource id #841, Object(Hirak\Prestissimo\Aspects\HttpGetRequest))
#1 /home/florian/workspace/qspot/vendor/hirak/prestissimo/src/CurlRemoteFilesystem.php(86): Hirak\Prestissimo\CurlRemoteFilesystem->fetch('api.github.com', 'https://api.git...', true, Array, Object(Closure))
#2 phar:///usr/bin/composer/src/Composer/Downloader/FileDownloader.php(136): Hirak\Prestissimo\CurlRemoteFilesystem->copy('api.github.com', 'https://api.git...', '/home/florian/w...', true, Array)
#3 phar:///usr/bin/composer/src/Composer/Downloader/FileDownloader.php(88): Composer\Downloader\FileDownloader->doDownload(Object(Composer\Package\CompletePackage), '/home/florian/w...', 'https://api.git...')
#4 phar:///usr/ in /home/florian/workspace/qspot/vendor/hirak/prestissimo/src/CurlRemoteFilesystem.php on line 73

Fatal error: Uncaught Error: Class 'Hirak\Prestissimo\OutputFile' not found in /home/florian/workspace/qspot/vendor/hirak/prestissimo/src/CurlRemoteFilesystem.php on line 73

Error: Class 'Hirak\Prestissimo\OutputFile' not found in /home/florian/workspace/qspot/vendor/hirak/prestissimo/src/CurlRemoteFilesystem.php on line 73

Call Stack:
    0.0006     354736   1. {main}() /usr/bin/composer:0
    0.0060     549136   2. require('phar:///usr/bin/composer/bin/composer') /usr/bin/composer:24
    0.0288    2586992   3. Composer\Console\Application->run() phar:///usr/bin/composer/bin/composer:43
    0.0308    2869128   4. Symfony\Component\Console\Application->run() phar:///usr/bin/composer/src/Composer/Console/Application.php:82
    0.0316    2938712   5. Composer\Console\Application->doRun() phar:///usr/bin/composer/vendor/symfony/console/Application.php:120
    0.0326    3005360   6. Symfony\Component\Console\Application->doRun() phar:///usr/bin/composer/src/Composer/Console/Application.php:147
    0.0328    3005360   7. Symfony\Component\Console\Application->doRunCommand() phar:///usr/bin/composer/vendor/symfony/console/Application.php:189
    0.0328    3005360   8. Symfony\Component\Console\Command\Command->run() phar:///usr/bin/composer/vendor/symfony/console/Application.php:838
    0.0331    3007632   9. Composer\Command\UpdateCommand->execute() phar:///usr/bin/composer/vendor/symfony/console/Command/Command.php:256
    0.1427    6827296  10. Composer\Installer->run() phar:///usr/bin/composer/src/Composer/Command/UpdateCommand.php:143
    0.1469    6926792  11. Composer\Installer->doInstall() phar:///usr/bin/composer/src/Composer/Installer.php:232
   36.7447  506513000  12. Composer\Installer\InstallationManager->execute() phar:///usr/bin/composer/src/Composer/Installer.php:603
   36.7447  506513000  13. Composer\Installer\InstallationManager->update() phar:///usr/bin/composer/src/Composer/Installer/InstallationManager.php:139
   36.7447  506513000  14. Composer\Installer\PluginInstaller->update() phar:///usr/bin/composer/src/Composer/Installer/InstallationManager.php:172
   36.7447  506513000  15. Composer\Installer\LibraryInstaller->update() phar:///usr/bin/composer/src/Composer/Installer/PluginInstaller.php:75
   36.7449  506512960  16. Composer\Installer\LibraryInstaller->updateCode() phar:///usr/bin/composer/src/Composer/Installer/LibraryInstaller.php:109
   36.7450  506513120  17. Composer\Downloader\DownloadManager->update() phar:///usr/bin/composer/src/Composer/Installer/LibraryInstaller.php:180
   36.7450  506513120  18. Composer\Downloader\FileDownloader->update() phar:///usr/bin/composer/src/Composer/Downloader/DownloadManager.php:253
   36.7628  506516520  19. Composer\Downloader\ArchiveDownloader->download() phar:///usr/bin/composer/src/Composer/Downloader/FileDownloader.php:199
   36.7629  506516600  20. Composer\Downloader\FileDownloader->download() phar:///usr/bin/composer/src/Composer/Downloader/ArchiveDownloader.php:35
   36.7631  506517016  21. Composer\Downloader\FileDownloader->doDownload() phar:///usr/bin/composer/src/Composer/Downloader/FileDownloader.php:88
   36.7706  506817424  22. Hirak\Prestissimo\CurlRemoteFilesystem->copy() phar:///usr/bin/composer/src/Composer/Downloader/FileDownloader.php:136
   36.7707  506818120  23. Hirak\Prestissimo\CurlRemoteFilesystem->fetch() /home/florian/workspace/qspot/vendor/hirak/prestissimo/src/CurlRemoteFilesystem.php:86
   36.7718  506823032  24. Hirak\Prestissimo\CurlRemoteFilesystem->Hirak\Prestissimo\{closure}() /home/florian/workspace/qspot/vendor/hirak/prestissimo/src/CurlRemoteFilesystem.php:173

```

**Comment by hirak** [comment]
Please try `--no-plugins` option.

```
$ composer update --no-plugins
```

**Comment by it-can** [comment]
I update this way:

```
composer global update -o
```
And get the same error
```
Updating dependencies (including require-dev)
  - Removing hirak/prestissimo (0.1.9)
  - Installing hirak/prestissimo (0.1.10)
    Downloading: Connecting...
Fatal error: Class 'Hirak\Prestissimo\OutputFile' not found in /root/.composer/vendor/hirak/prestissimo/src/CurlRemoteFilesystem.php on line 73
```

When I run the update command the second time it does not throw this error...

**Comment by docteurklein** [comment]
I don't know if it helps, but puli seems to handle class loading manually here:

https://github.com/puli/composer-plugin/blob/164ad0201ef8a242521145173575efb271cef237/src/PuliPlugin.php#L70-L78

**Comment by hirak** [comment]
@docteurklein Thank you for your help. This is a difficult problem.
I fixed at #50. But this feature enables at `prestissimo >= 0.1.11`.

You should reinstall prestissimo once.

```bash
$ composer remove --dev hirak/prestissimo
$ composer require --dev hirak/prestissimo
```
